yt_dlp/extractor/taptap.py

```diff
@@ -0,0 +1,148 @@
+            "title": "《崩坏：星穹铁道》黄泉角色PV——「你的颜色」",
+            "description": "md5:2c81245da864428c904d53ae4ad2182b",
+            "timestamp": 1711425600,
+            "upload_date": "20240326",
+            "uploader": "崩坏：星穹铁道",
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '4006511',
+                "ext": "mp4",
+                "title": "《崩坏：星穹铁道》黄泉角色PV——「你的颜色」",
+                "description": "md5:2c81245da864428c904d53ae4ad2182b",
+                "duration": 173,
+                "timestamp": 1711425600,
+                "upload_date": "20240326",
+                "uploader": "崩坏：星穹铁道",
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }, {
+        'url': 'https://www.taptap.cn/app/168332',
+        'info_dict': {
+            'id': 'app_168332',
+            "title": "原神",
+            "description": "md5:e345f39a5fea5de2a46923f70d5f76ab",
+        },
+        'playlist_count': 2,
+        'playlist': [{
+            'info_dict': {
+                'id': '4058443',
+                "ext": "mp4",
+                "title": "原神",
+                "description": "md5:e345f39a5fea5de2a46923f70d5f76ab",
+                "duration": 26,
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }, {
+            'info_dict': {
+                'id': '4058462',
+                "ext": "mp4",
+                "title": "原神",
+                "description": "md5:e345f39a5fea5de2a46923f70d5f76ab",
+                "duration": 295,
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+
+    }]
+
+    def _deserialize_nuxt_data(self, serialized_nuxt):
+        for row in serialized_nuxt:
+            if isinstance(row, dict):
+                for key, value_or_ref in row.items():
+                    if isinstance(value_or_ref, int):
+                        row[key] = serialized_nuxt[value_or_ref]
+            elif isinstance(row, list):
+                for index, value_or_ref in tuple(enumerate(row)):
+                    if isinstance(value_or_ref, int):
+                        row[index] = serialized_nuxt[value_or_ref]
+        return serialized_nuxt[0]
+
+    def _extract_video(self, video_id, x_ua):
+        data = self._download_json(
+            'https://www.taptap.cn/webapiv2/video-resource/v1/multi-get', video_id,
+            query={'video_ids': video_id, 'X-UA': x_ua})
+
+        video = traverse_obj(data, ('data', 'list', 0, {
+            'id': ('video_id', {str_or_none}),
+            'url': ('play_url', ('url', 'url_h265'), {url_or_none}),
+            'duration': ('info', 'duration', {int_or_none}),
+            'thumbnail': ('thumbnail', ('original_url', 'url'), {url_or_none}),
+        }), get_all=False)
+        if '.m3u8' in video['url']:
+            video['formats'] = self._extract_m3u8_formats(video.pop('url'), video_id)
```

Comment:
I would extract both the AVC and HEVC formats every time. I think that additional request is worth it, or is there any reason like excessive rate limiting not to do this?